### PR TITLE
[feat]: [CDS-43127]: Replace copy with download command for winrm

### PR DESCRIPTION
**Pull request by vladimir-peric**
## Describe your changes
Download command unit is introduced and is used instead of copy command unit in auto-generated yaml snippets for winrm pipelines.

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/37844)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/37844)

**Comment by vladimir-peric** [comment]
trigger messagecheck

**Review by ivan-83 — CHANGES_REQUESTED** [review]
Well done with minor comments.

> Reviewed 20 of 29 files at r1, 10 of 10 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37844#-:-NCg-xQI5h9cKwCm7WjN:b-hvtfbu)* status: 1 change requests. 0 of 1 approvals obtained. 

___
*[`125-cd-nextgen/src/main/resources/snippets/Pipelines/execution/ssh/script/setup-runtime-paths-script-powershell.yaml` line 2 at r2](https://reviewable.io/reviews/harness/harness-core/37844#-NCg-QCa8Bv2Zn66MXj-:-NCg-QCa8Bv2Zn66MXj0:b-l9qcnh) ([raw file](https://github.com/harness/harness-core/blob/2bcd3f79d5731e5f34021578b2a3d6754da586f8/125-cd-nextgen/src/main/resources/snippets/Pipelines/execution/ssh/script/setup-runtime-paths-script-powershell.yaml#L2)):*
> ```YAML
> # Creates destination folders if they don't exist
>                               $FolderName = "%USERPROFILE%\<+service.name>\<+env.name>\runtime"
> ```

Do we need runtime, backup, and staging directories or only "%USERPROFILE%\<+service.name>\<+env.name> ?
___
*[`125-cd-nextgen/src/main/resources/snippets/Pipelines/execution/ssh/script/setup-runtime-paths-script-rolling-powershell.yaml` line 2 at r2](https://reviewable.io/reviews/harness/harness-core/37844#-NCg-bmZ1RJoawg4U--G:-NCg-bmZ1RJoawg4U--H:b-l9qcnh) ([raw file](https://github.com/harness/harness-core/blob/2bcd3f79d5731e5f34021578b2a3d6754da586f8/125-cd-nextgen/src/main/resources/snippets/Pipelines/execution/ssh/script/setup-runtime-paths-script-rolling-powershell.yaml#L2)):*
> ```YAML
> # Creates destination folders if they don't exist
>                                     $FolderName = "%USERPROFILE%\<+service.name>\<+env.name>\runtime"
> ```

Do we need runtime, backup, and staging directories or only "%USERPROFILE%\<+service.name>\<+env.name> ?

**Review by ivan-83 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37844)* status: 1 change requests. 0 of 1 approvals obtained. 

___
*[`125-cd-nextgen/src/main/resources/snippets/Pipelines/execution/ssh/script/setup-runtime-paths-script-powershell.yaml` line 3 at r2](https://reviewable.io/reviews/harness/harness-core/37844#-NCg0w79CJxgd4VTQmpW:-NCg0w79CJxgd4VTQmpX:b87txxg) ([raw file](https://github.com/harness/harness-core/blob/2bcd3f79d5731e5f34021578b2a3d6754da586f8/125-cd-nextgen/src/main/resources/snippets/Pipelines/execution/ssh/script/setup-runtime-paths-script-powershell.yaml#L3)):*
> ```YAML
> # Creates destination folders if they don't exist
>                               $FolderName = "%USERPROFILE%\<+service.name>\<+env.name>\runtime"
>                               If (!(Test-Path $FolderName)) {
> ```

I think it is better to use directory instead of folder terminology on win OS. I mean use $DirectoryName instead of $FolderName

**Review by acasian-harness — APPROVED** [review]
> Reviewed 20 of 29 files at r1, 10 of 10 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37844)* status: 1 change requests. 2. unresolved discussions. 

___
*[`125-cd-nextgen/src/main/resources/snippets/Pipelines/execution/ssh/script/setup-runtime-paths-script-powershell.yaml` line 2 at r2](https://reviewable.io/reviews/harness/harness-core/37844#-NCg-QCa8Bv2Zn66MXj-:-NCko9L28gL2R7UpeL3c:bi1kj07) ([raw file](https://github.com/harness/harness-core/blob/2bcd3f79d5731e5f34021578b2a3d6754da586f8/125-cd-nextgen/src/main/resources/snippets/Pipelines/execution/ssh/script/setup-runtime-paths-script-powershell.yaml#L2)):*
<details><summary><i>Previously, ivan-83 (Ivan Mijailovic) wrote…</i></summary><blockquote>

Do we need runtime, backup, and staging directories or only "%USERPROFILE%\<+service.name>\<+env.name> ?
</blockquote></details>

nope, only `%USERPROFILE%<+service.name><+env.name>`
___
*[`125-cd-nextgen/src/main/resources/snippets/Pipelines/execution/ssh/script/setup-runtime-paths-script-rolling-powershell.yaml` line 2 at r2](https://reviewable.io/reviews/harness/harness-core/37844#-NCg-bmZ1RJoawg4U--G:-NCkoL-5B16LDoyT4bzN:bb7lgjf) ([raw file](https://github.com/harness/harness-core/blob/2bcd3f79d5731e5f34021578b2a3d6754da586f8/125-cd-nextgen/src/main/resources/snippets/Pipelines/execution/ssh/script/setup-runtime-paths-script-rolling-powershell.yaml#L2)):*
<details><summary><i>Previously, ivan-83 (Ivan Mijailovic) wrote…</i></summary><blockquote>

Do we need runtime, backup, and staging directories or only "%USERPROFILE%\<+service.name>\<+env.name> ?
</blockquote></details>

nope

**Comment by vladimir-peric** [comment]
*[`125-cd-nextgen/src/main/resources/snippets/Pipelines/execution/ssh/script/setup-runtime-paths-script-powershell.yaml` line 3 at r2](https://reviewable.io/reviews/harness/harness-core/37844#-NCg0w79CJxgd4VTQmpW:-NCnjnOE5pKOSwdhggN9:b-so00pk) ([raw file](https://github.com/harness/harness-core/blob/2bcd3f79d5731e5f34021578b2a3d6754da586f8/125-cd-nextgen/src/main/resources/snippets/Pipelines/execution/ssh/script/setup-runtime-paths-script-powershell.yaml#L3)):*
<details><summary><i>Previously, ivan-83 (Ivan Mijailovic) wrote…</i></summary><blockquote>

I think it is better to use directory instead of folder terminology on win OS. I mean use $DirectoryName instead of $FolderName 
</blockquote></details>

I think folder more aligns with Windows terminology and directory with linux, however these can be used interchangeably.

**Comment by vladimir-peric** [comment]
*[`125-cd-nextgen/src/main/resources/snippets/Pipelines/execution/ssh/script/setup-runtime-paths-script-powershell.yaml` line 2 at r2](https://reviewable.io/reviews/harness/harness-core/37844#-NCg-QCa8Bv2Zn66MXj-:-NCnlsyCC_oBBuS4fbF1:b-udze8u) ([raw file](https://github.com/harness/harness-core/blob/2bcd3f79d5731e5f34021578b2a3d6754da586f8/125-cd-nextgen/src/main/resources/snippets/Pipelines/execution/ssh/script/setup-runtime-paths-script-powershell.yaml#L2)):*
<details><summary><i>Previously, acasian-harness (Alexandru Casian) wrote…</i></summary><blockquote>

nope, only `%USERPROFILE%<+service.name><+env.name>`
</blockquote></details>

We actually might want to keep %USERPROFILE%\<+service.name>\<+env.name>\runtime as this is used as target location in some scripts, so we don't need to check if it exists and create it in scripts, eg. create-apppool-script-powershell.yaml

**Review by ivan-83 — CHANGES_REQUESTED** [review]
> Reviewed 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37844)* status: 1 change requests. 2. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`125-cd-nextgen/src/main/resources/snippets/Pipelines/execution/ssh/script/setup-runtime-paths-script-powershell.yaml` line 2 at r2](https://reviewable.io/reviews/harness/harness-core/37844#-NCg-QCa8Bv2Zn66MXj-:-NCpgGSH0X9xL4CeEmvc:b-xz4jzi) ([raw file](https://github.com/harness/harness-core/blob/2bcd3f79d5731e5f34021578b2a3d6754da586f8/125-cd-nextgen/src/main/resources/snippets/Pipelines/execution/ssh/script/setup-runtime-paths-script-powershell.yaml#L2)):*
<details><summary><i>Previously, vladimir-peric (Vladimir Peric) wrote…</i></summary><blockquote>

We actually might want to keep %USERPROFILE%\<+service.name>\<+env.name>\runtime as this is used as target location in some scripts, so we don't need to check if it exists and create it in scripts, eg. create-apppool-script-powershell.yaml
</blockquote></details>

Ok, could you please double-check in CG doe we have a runtime path in Create Apppool Script, if not let's remove it and keep only %USERPROFILE%/<+service.name><+env.name>. 
___
*[`125-cd-nextgen/src/main/resources/snippets/Pipelines/execution/ssh/script/setup-runtime-paths-script-powershell.yaml` line 3 at r2](https://reviewable.io/reviews/harness/harness-core/37844#-NCg0w79CJxgd4VTQmpW:-NCpiFGN8hD2v35opFVC:b-9bg3pj) ([raw file](https://github.com/harness/harness-core/blob/2bcd3f79d5731e5f34021578b2a3d6754da586f8/125-cd-nextgen/src/main/resources/snippets/Pipelines/execution/ssh/script/setup-runtime-paths-script-powershell.yaml#L3)):*
<details><summary><i>Previously, vladimir-peric (Vladimir Peric) wrote…</i></summary><blockquote>

I think folder more aligns with Windows terminology and directory with linux, however these can be used interchangeably. 
</blockquote></details>

Yes, I remember, you are right, my mistake, directories come from UNIX because of indexes.

**Comment by vladimir-peric** [comment]
*[`125-cd-nextgen/src/main/resources/snippets/Pipelines/execution/ssh/script/setup-runtime-paths-script-powershell.yaml` line 2 at r2](https://reviewable.io/reviews/harness/harness-core/37844#-NCg-QCa8Bv2Zn66MXj-:-NCpsTKRD35_cgY-eGwu:b9i5p3s) ([raw file](https://github.com/harness/harness-core/blob/2bcd3f79d5731e5f34021578b2a3d6754da586f8/125-cd-nextgen/src/main/resources/snippets/Pipelines/execution/ssh/script/setup-runtime-paths-script-powershell.yaml#L2)):*
<details><summary><i>Previously, ivan-83 (Ivan Mijailovic) wrote…</i></summary><blockquote>

Ok, could you please double-check in CG doe we have a runtime path in Create Apppool Script, if not let's remove it and keep only %USERPROFILE%/<+service.name><+env.name>. 
</blockquote></details>

Yes, in cg in Create AppPool script there is a line: Copy-Item -Path "$userProfile\\${app.name}\\${serviceName}\\${env.name}\\runtime\\*.config" -Destination $appPhysicalDirectory -Force

File: 125-cd-nextgen/src/main/resources/snippets/Pipelines/execution/ssh/script/setup-runtime-paths-script-powershell.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Creates destination folders if they don't exist
+                              $FolderName = "%USERPROFILE%\<+service.name>\<+env.name>\runtime"
+                              If (!(Test-Path $FolderName)) {
+                                Write-Host "Creating folder" $FolderName
+                                New-Item $FolderName -ItemType Directory
+                              }
+                              $FolderName = "%USERPROFILE%\<+service.name>\<+env.name>\backup"
+                              If (!(Test-Path $FolderName)) {
```
Comment (filip5rovic) [inline 125-cd-nextgen/src/main/resources/snippets/Pipelines/execution/ssh/script/setup-runtime-paths-script-powershell.yaml]: The code format looks wrong here. Check indentation for if block

File: 125-cd-nextgen/src/main/resources/snippets/Pipelines/execution/ssh/script/setup-runtime-paths-script-powershell.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+                              Write-Host "Creating folder" $FolderName
+                              New-Item $FolderName -ItemType Directory
+                              }
+                              $FolderName = "%USERPROFILE%\<+service.name>\<+env.name>\staging"
+                              If (!(Test-Path $FolderName)) {
+                              Write-Host "Creating folder" $FolderName
```
Comment (filip5rovic) [inline 125-cd-nextgen/src/main/resources/snippets/Pipelines/execution/ssh/script/setup-runtime-paths-script-powershell.yaml]: The code format looks wrong here. Check indentation for if block

File: 125-cd-nextgen/src/main/resources/snippets/Pipelines/execution/ssh/script/setup-runtime-paths-script-rolling-powershell.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# Creates destination folders if they don't exist
+                                    $FolderName = "%USERPROFILE%\<+service.name>\<+env.name>\runtime"
+                                    If (!(Test-Path $FolderName)) {
+                                      Write-Host "Creating folder" $FolderName
+                                      New-Item $FolderName -ItemType Directory
+                                    }
+                                    $FolderName = "%USERPROFILE%\<+service.name>\<+env.name>\backup"
+                                    If (!(Test-Path $FolderName)) {
+                                    Write-Host "Creating folder" $FolderName
```
Comment (filip5rovic) [inline 125-cd-nextgen/src/main/resources/snippets/Pipelines/execution/ssh/script/setup-runtime-paths-script-rolling-powershell.yaml]: The code format looks wrong here. Check indentation for if block

File: 125-cd-nextgen/src/main/resources/snippets/Pipelines/execution/ssh/script/setup-runtime-paths-script-rolling-powershell.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+                                    New-Item $FolderName -ItemType Directory
+                                    }
+                                    $FolderName = "%USERPROFILE%\<+service.name>\<+env.name>\staging"
+                                    If (!(Test-Path $FolderName)) {
+                                    Write-Host "Creating folder" $FolderName
```
Comment (filip5rovic) [inline 125-cd-nextgen/src/main/resources/snippets/Pipelines/execution/ssh/script/setup-runtime-paths-script-rolling-powershell.yaml]: The code format looks wrong here. Check indentation for if block

File: 125-cd-nextgen/src/main/resources/snippets/Pipelines/execution/ssh/script/setup-runtime-paths-script-powershell.yaml (hunk 1 above)
Comment (filip5rovic) [reply 125-cd-nextgen/src/main/resources/snippets/Pipelines/execution/ssh/script/setup-runtime-paths-script-powershell.yaml]: Looks like this code block has been deleted. Resolving comments

**Review by filip5rovic — COMMENTED** [review]
> Reviewed 20 of 29 files at r1, 8 of 10 files at r2, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37844)* status: 1 change requests. 2. unresolved discussions. 0 of 1 approvals obtained.

**Review by filip5rovic — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37844#-:-NCskurQ4-q-F_X7mVUg:bnfp4nl)* status: 1 change requests. 2. unresolved discussions.

**Comment by vladimir-peric** [comment]
*[`125-cd-nextgen/src/main/resources/snippets/Pipelines/execution/ssh/script/setup-runtime-paths-script-powershell.yaml` line 8 at r2](https://reviewable.io/reviews/harness/harness-core/37844#-NCdOLDyVwrdc9mkPzPn-r2-8:-NCskoe68oRFizvA8b3b:b-i7bsbm) ([raw file](https://github.com/harness/harness-core/blob/2bcd3f79d5731e5f34021578b2a3d6754da586f8/125-cd-nextgen/src/main/resources/snippets/Pipelines/execution/ssh/script/setup-runtime-paths-script-powershell.yaml#L8)):*
<details><summary><i>Previously, filip5rovic (Filip Petrovic) wrote…</i></summary><blockquote>

Looks like this code block has been deleted. Resolving comments
</blockquote></details>


This is done intentionally, as this script is inserted into a yaml, and the indentation matches the indentation in the resulting yaml.

**Pull request closed without merging**